Raw JSON displayed when adding category to array of strings

When you use an array of strings and the tagging feature of the Multiselect component, it shows you raw JSON code in the dropdown after you searched for a category / want to add a new one.

This is how it looks like:
<img width="331" alt="58124057-95bb2600-7c0d-11e9-86b0-ab42cbcff739" src="https://user-images.githubusercontent.com/2496460/58175161-c8adfa00-7c9f-11e9-9be0-3e1bb84aab45.png">

Corresponding demo code:
```
<template>
	<div>
		<Multiselect
			v-model="categories"
			:multiple="true"
			:searchable="true"
			:options="categories"
			:placeholder="'Select categories'"
			:taggable="true"
			:tag-placeholder="'Add this as a new category'"
			:close-on-select="false"
			class="multiselect-vue"
			@input="updateCategories"
			@tag="updateCategory"
		/>
	</div>
</template>

<script>
import { Multiselect } from 'nextcloud-vue'

export default {
	components: {
		Multiselect,
	},
	data: function() {
		return {
			categories: ['a', 'b', 'c'],
		}
	},
	methods: {
		/**
		 * Sets the categories of a task
		 *
		 * @param {Array} categories The new categories
		 */
		updateCategories: function(categories) {
			console.debug('Set categories to ' + categories)
		},

		/**
		 * Adds a category to the list of categories
		 *
		 * @param {String} category The name of the category to add
		 */
		updateCategory: function(category) {
			console.debug('Add categoriy ' + category)
		},
	}
}
</script>
```

The same code works just fine with the upstream `vue-multiselect`:
<img width="332" alt="noerror" src="https://user-images.githubusercontent.com/2496460/58175202-e54a3200-7c9f-11e9-9c1e-b73b8be2985e.png">

Code (only the import statement is changed):
```
<template>
	<div>
		<Multiselect
			v-model="categories"
			:multiple="true"
			:searchable="true"
			:options="categories"
			:placeholder="'Select categories'"
			:taggable="true"
			:tag-placeholder="'Add this as a new category'"
			:close-on-select="false"
			class="multiselect-vue"
			@input="updateCategories"
			@tag="updateCategory"
		/>
	</div>
</template>

<script>
import Multiselect from 'vue-multiselect'

export default {
	components: {
		Multiselect,
	},
	data: function() {
		return {
			categories: ['a', 'b', 'c'],
		}
	},
	methods: {
		/**
		 * Sets the categories of a task
		 *
		 * @param {Array} categories The new categories
		 */
		updateCategories: function(categories) {
			console.debug('Set categories to ' + categories)
		},

		/**
		 * Adds a category to the list of categories
		 *
		 * @param {String} category The name of the category to add
		 */
		updateCategory: function(category) {
			console.debug('Add categoriy ' + category)
		},
	}
}
</script>
```

Also see https://github.com/nextcloud/tasks/issues/416 where this was initially reported.

Thanks @juliushaertl. But when I do this, I get a lot of errors for `ncvuecomponents.js` when running `npm run watch` in the Tasks folder:

```
ERROR in ../nextcloud-vue/dist/ncvuecomponents.js
Module Error (from ./node_modules/eslint-loader/index.js):

/var/www/nextcloud/apps/nextcloud-vue/dist/ncvuecomponents.js
      2:2      error  Expected { after 'if' condition                           
                                                                                
                                                                                
                                                                                
                                                                                
                                                     curly
      2:2      error  Expected space(s) after "if"                              
                                                                                
                                                                                
                                                                                
                                                                                
                                                     keyword-spacing
      3:36     error  Strings must use singlequote                              
                                                                                
                                                                                
                                                                                
                                                                                
                                                     quotes
      3:43     error  Extra semicolon                                           
                                                                                
                                                                                
                                                                                
                                                                                                                                     semi
      4:7      error  Expected space(s) after "if"                                                                                                                                                                                                                                                                                                                                                                                                                   keyword-spacing
      4:7      error  Expected { after 'if' condition                                                                                                                                                                                                                                                                                                                                                                                                                curly
      4:42     error  'define' is not defined                                                                                                                                                                                                                                                                                                                                                                                                                        no-undef
      5:3      error  'define' is not defined

...
```

--- a/src/components/Multiselect/EllipsisedOption.vue
+++ b/src/components/Multiselect/EllipsisedOption.vue
@@ -42,13 +42,7 @@ export default {
 	},
 	computed: {
 		name() {
-			if (this.label) {
-				return this.option[this.label]
-			}
-			if (this.option.isTag) {
-				return this.option.label
-			}
-			return this.option
+			return this.$parent.getOptionLabel(this.option)
 		},
 		needsTruncate() {
 			return this.name && this.name.length >= 10

--- a/src/components/Multiselect/EllipsisedOption.vue
+++ b/src/components/Multiselect/EllipsisedOption.vue
@@ -45,6 +45,9 @@ export default {
 			if (this.label) {
 				return this.option[this.label]
 			}
+			if (this.option.isTag) {
+				return this.option.label
+			}
 			return this.option
 		},
 		needsTruncate() {

